posts/2020/csv-downloader-in-react.md

```diff
@@ -0,0 +1,41 @@
+.. title: CSV Downloader in React
+.. slug: csv-downloader-in-react
+.. date: 2020-07-17 13:17:20 UTC+05:30
+.. tags: ReactJS
+.. link:
+.. description:
+.. type: text
+.. summary: In this post, we will check how the UI data can be downloaded in CSV format. This functionality can be achieved on the back-end (Django REST) itself using 'Content-Disposition' header but we will prefer front-end approach for implementation and also explain why so.
+
+# Why CSV?
+
+Data is main element of any application. We won't be using many of these apps without data. Data transfer (between computers or apps) increases the usability even further as the same data can be used for different purposes.
+
+eg. We have the **_Library_** app that shows library books & members. We see all the books in table format. But, wouldn't it be great if we can export the books data in some 'standard data type'. Date type which is 'widely used' & simple to convert, so it can be transferred & used by many computers, applications?
+
+CSV (Comma Separated Values) is widely used data type for data exchange between computers. It stores data in plain text using comma as separator. We can convert & download our books data (json) as csv file so it can be used for different purposes by different computers, applications.
+
+# How CSV?
+
+Explain how can we implement csv download functionality on both-ends
+
+## Back-end Approach
```

Comment:
Looks like - you have not understood the problem. Backend alone does not work on Ajax requests. That is the main important reason! We simply don't do it in front-end because it's cool. There were specific problems that we had to address and there was not one single document one could refer and hence we are writing this post.